data/plugins/entity/interaction/object.rb

```diff
@@ -0,0 +1,6 @@
+# Catch all object actions and turn the player to
+# the position of the object
+
+on :message, :object_action do |player, message|
+  player.turn_to message.position
```

Comment:
From Client#processClickedObject

```java
            int surroundings = definition.getSurroundings();
            if (orientation != 0) {
                surroundings = (surroundings << orientation & 0xF) + (surroundings >> 4 - orientation);
            }
```

The flag is passed to the walk method and checked in CollisionMap#reachedObejct

```java
    public boolean reachedObject(int x, int y, int finalX, int finalY, int height, int surroundings, int width) {
        int maxX = finalX + width - 1;
        int maxY = finalY + height - 1;

        if (x >= finalX && x <= maxX && y >= finalY && y <= maxY) {
            return true;
        } else if (x == finalX - 1 && y >= finalY && y <= maxY
                && (adjacencies[x - xOffset][y - yOffset] & WALL_EAST) == 0 && (surroundings & WALL_EAST) == 0) {
            return true;
        } else if (x == maxX + 1 && y >= finalY && y <= maxY && (adjacencies[x - xOffset][y - yOffset] & WALL_WEST) == 0
                && (surroundings & WALL_NORTH) == 0) {
            return true;
        } else if (y == finalY - 1 && x >= finalX && x <= maxX
                && (adjacencies[x - xOffset][y - yOffset] & WALL_NORTH) == 0 && (surroundings & WALL_NORTHEAST) == 0) {
            return true;
        }

        return y == maxY + 1 && x >= finalX && x <= maxX && (adjacencies[x - xOffset][y - yOffset] & WALL_SOUTH) == 0
                && (surroundings & WALL_NORTHWEST) == 0;
    }
```